### PR TITLE
feat: add release-it

### PR DESCRIPTION
**Pull request by freeboub**
Integrate release-it tool for new release management

Next version can be generated with:
npx release-it --preRelease=alpha

should address: https://github.com/react-native-video/react-native-video/issues/3333

### CODE REVIEW

**Comment by evoactivity** [comment]
I've been thinking about adding changesets here, it works a lot better for community projects as you don't have to enforce every PR to follow conventional commits, it makes keeping a changelog very simple as peoples changesets get added automatically. It can be completely automated through github actions so anyone with merge access on the repo will be able to make a release. Releases can goto github and npm.

https://github.com/changesets/changesets

I have a repo using automated changesets here if you would like to see how I have it setup
https://github.com/ektotv/playlist

**Comment by KrzysztofMoch** [comment]
@evoactivity Thanks for your feedback! 
But I think that for us it will be much easier to use [release-it](https://github.com/release-it/release-it) with [changelog plugin](https://github.com/release-it/conventional-changelog). To enforce that every PR follow conventional commits we can use [github action](https://github.com/amannn/action-semantic-pull-request). We (I and @freeboub) have already used release-it so it will be easier for us to configurate and use it

File: package.json [hunk 1]
```diff
@@ -41,6 +43,7 @@
         "prepare": "yarn build",
         "xbasic": "yarn --cwd examples/basic",
         "docs": "yarn --cwd docs build",
+        "release": "release-it",
         "test": "echo no test available"
     },
     "files": [
```
Comment (KrzysztofMoch) [inline package.json:49]: we should make sure that next release won't have docs and example 😅
```suggestion
    "files": [
        ...
        "!docs",
        "!examples",
        ...
    ]
```

**Review by KrzysztofMoch — CHANGES_REQUESTED** [review]
we need update `package.json` to exclude `docs` and `examples` and we are ready to go

**Comment by freeboub** [comment]
> we need update `package.json` to exclude `docs` and `examples` and we are ready to go

I added, but I think this is not mandatory, all the !XXX are subfolders of included folders.

**Comment by freeboub** [comment]
> I've been thinking about adding changesets here, it works a lot better for community projects as you don't have to enforce every PR to follow conventional commits, it makes keeping a changelog very simple as peoples changesets get added automatically. It can be completely automated through github actions so anyone with merge access on the repo will be able to make a release. Releases can goto github and npm.
> 
> https://github.com/changesets/changesets
> 
> I have a repo using automated changesets here if you would like to see how I have it setup https://github.com/ektotv/playlist

Thank you for the clue, I agree it would be interesting to have github integration, but I had this release-it integration under the hand, it was very easy.

**Review by KrzysztofMoch — APPROVED** [review]
LGTM 👍

**Pull request merged**